Crash recording audio

* record
* stop recording
* do not save
* press the tick before pressing 'save'



on `main`: 3cf54e7affdd24c87e1aaa02ed55369cdee1cf91

<details>

```
10:57:23.980 AnkiDroidApp$onCreate   com.ichi2.anki.debug                 I  NoteEditor::onPause
10:57:23.990 ViewRootIm...w:8089e18] com.ichi2.anki.debug                 I  registerCallbackForPendingTransactions
10:57:24.008 Themes                  com.ichi2.anki.debug                 I  Setting theme to LIGHT
10:57:24.011 AnkiDroidApp$onCreate   com.ichi2.anki.debug                 I  MultimediaEditFieldActivity::onCreate
10:57:24.038 Multimedia...ldActivity com.ichi2.anki.debug                 D  recreateEditingUi()
10:57:24.039 Multimedia...ionHandler com.ichi2.anki.debug                 D  onPreFieldControllerReplacement
10:57:24.120 AudioRecor...Controller com.ichi2.anki.debug                 D  Creating media player for playback
10:57:24.120 MediaPlayer-JNI         com.ichi2.anki.debug                 V  native_setup
10:57:24.120 MediaPlayerNative       com.ichi2.anki.debug                 V  constructor
10:57:24.123 MediaPlayerNative       com.ichi2.anki.debug                 V  setListener
10:57:24.124 MediaPlayer-JNI         com.ichi2.anki.debug                 V  get_session_id()
10:57:24.133 SensorManager           com.ichi2.anki.debug                 D  registerListener :: 1, LSM6DSO Accelerometer, 200000, 0, 
10:57:24.134 Multimedia...ionHandler com.ichi2.anki.debug                 D  onPostUICreation. State: 0
10:57:24.136 AnkiDroidApp$onCreate   com.ichi2.anki.debug                 I  MultimediaEditFieldActivity::onStart
10:57:24.139 AnkiDroidApp$onCreate   com.ichi2.anki.debug                 I  MultimediaEditFieldActivity::onResume
10:57:24.139 UsageAnalytics          com.ichi2.anki.debug                 D  sendAnalyticsScreenView(): MultimediaEditFieldActivity
10:57:24.139 UsageAnalytics          com.ichi2.anki.debug                 D  getOptIn() status: false
10:57:24.141 NativeCust...ncyManager com.ichi2.anki.debug                 D  [NativeCFMS] BpCustomFrequencyManager::BpCustomFrequencyManager()
10:57:24.145 InsetsController        com.ichi2.anki.debug                 D  onStateChanged: host=com.ichi2.anki.debug/com.ichi2.anki.multimediacard.activity.MultimediaEditFieldActivity from=android.view.ViewRootImpl.setView:1717 state=InsetsState: {mDisplayFrame=Rect(0, 0 - 1080, 2400), mDisplayCutout=DisplayCutout{insets=Rect(0, 80 - 0, 0) waterfall=Insets{left=0, top=0, right=0, bottom=0} boundingRect={Bounds=[Rect(0, 0 - 0, 0), Rect(510, 0 - 570, 80), Rect(0, 0 - 0, 0), Rect(0, 0 - 0, 0)]} cutoutPathParserInfo={CutoutPathParserInfo{displayWidth=1080 displayHeight=2400 physicalDisplayWidth=1080 physicalDisplayHeight=2400 density={3.0} cutoutSpec={M 0,0 H -10 V 26.66666666666667 H 10 V 0 H 0 Z @dp} rotation={0} scale={1.0} physicalPixelDisplaySizeRatio={1.0}}}}, mRoundedCorners=RoundedCorners{[RoundedCorner{position=TopLeft, radius=0, center=Point(0, 0)}, RoundedCorner{position=TopRight, radius=0, center=Point(0, 0)}, RoundedCorner{position=BottomRight, radius=0, center=Point(0, 0)}, RoundedCorner{position=BottomLeft, radius=0, center=Point(0, 0)}]}  mRoundedCornerFrame=Rect(0, 0 - 1080, 2400), mPrivacyIndicatorBounds=PrivacyIndicatorBounds {static bounds=Rect(948, 0 - 1080, 80) rotation=0}, mDisplayShape=DisplayShape{ spec=-1893857183 displayWidth=1080 displayHeight=2400 physicalPixelDisplaySizeRatio=1.0 rotation=0 offsetX=0 offsetY=0 scale=1.0}, mSources= { InsetsSource: {c06f0001 mType=navigationBars mFrame=[0,2256][1080,2400] mVisible=true mFlags=[SUPPRESS_SCRIM]}, InsetsSource: {c06f0004 mType=systemGestures mFrame=[0,0][90,2400] mVisible=true mFlags=[]}, InsetsSource: {c06f0005 mType=mandatorySystemGestures mFrame=[0,2304][1080,2400] mVisible=true mFlags=[]}, InsetsSource: {c06f0006 mType=tappableElement mFrame=[0,0][0,0] mVisible=true mFlags=[]}, InsetsSource: {c06f0024 mType=systemGestures mFrame=[990,0][1080,2400] mVisible=true mFlags=[]}, InsetsSource: {e9c60000 mType=statusBars mFrame=[0,0][1080,80] mVisible=true mFlags=[]}, InsetsSource: {e9c60005 mType=mandatorySystemGestures mFrame=[0,0][1080,116] mVisible=true mFlags=[]}, InsetsSource: {e9c60006 mType=tappableElement mFrame=[0,0][1080,80] mVisible=true mFlags=[]}, InsetsSource: {27 mType=displayCutout mFrame=[0,0][1080,80] mVisible=true mFlags=[]} }
10:57:24.149 ViewRootIm...w:8089e18] com.ichi2.anki.debug                 I  dispatchDetachedFromWindow
10:57:24.174 InsetsController        com.ichi2.anki.debug                 D  onStateChanged: host=com.ichi2.anki.debug/com.ichi2.anki.multimediacard.activity.MultimediaEditFieldActivity from=android.view.ViewRootImpl.relayoutWindow:9940 state=InsetsState: {mDisplayFrame=Rect(0, 0 - 1080, 2400), mDisplayCutout=DisplayCutout{insets=Rect(0, 80 - 0, 0) waterfall=Insets{left=0, top=0, right=0, bottom=0} boundingRect={Bounds=[Rect(0, 0 - 0, 0), Rect(510, 0 - 570, 80), Rect(0, 0 - 0, 0), Rect(0, 0 - 0, 0)]} cutoutPathParserInfo={CutoutPathParserInfo{displayWidth=1080 displayHeight=2400 physicalDisplayWidth=1080 physicalDisplayHeight=2400 density={3.0} cutoutSpec={M 0,0 H -10 V 26.66666666666667 H 10 V 0 H 0 Z @dp} rotation={0} scale={1.0} physicalPixelDisplaySizeRatio={1.0}}}}, mRoundedCorners=RoundedCorners{[RoundedCorner{position=TopLeft, radius=0, center=Point(0, 0)}, RoundedCorner{position=TopRight, radius=0, center=Point(0, 0)}, RoundedCorner{position=BottomRight, radius=0, center=Point(0, 0)}, RoundedCorner{position=BottomLeft, radius=0, center=Point(0, 0)}]}  mRoundedCornerFrame=Rect(0, 0 - 1080, 2400), mPrivacyIndicatorBounds=PrivacyIndicatorBounds {static bounds=Rect(948, 0 - 1080, 80) rotation=0}, mDisplayShape=DisplayShape{ spec=-1893857183 displayWidth=1080 displayHeight=2400 physicalPixelDisplaySizeRatio=1.0 rotation=0 offsetX=0 offsetY=0 scale=1.0}, mSources= { InsetsSource: {c06f0001 mType=navigationBars mFrame=[0,2355][1080,2400] mVisible=true mFlags=[SUPPRESS_SCRIM]}, InsetsSource: {c06f0004 mType=systemGestures mFrame=[0,0][90,2400] mVisible=true mFlags=[]}, InsetsSource: {c06f0005 mType=mandatorySystemGestures mFrame=[0,2304][1080,2400] mVisible=true mFlags=[]}, InsetsSource: {c06f0006 mType=tappableElement mFrame=[0,0][0,0] mVisible=true mFlags=[]}, InsetsSource: {c06f0024 mType=systemGestures mFrame=[990,0][1080,2400] mVisible=true mFlags=[]}, InsetsSource: {e9c60000 mType=statusBars mFrame=[0,0][1080,80] mVisible=true mFlags=[]}, InsetsSource: {e9c60005 mType=mandatorySystemGestures mFrame=[0,0][1080,116] mVisible=true mFlags=[]}, InsetsSource: {e9c60006 mType=tappableElement mFrame=[0,0][1080,80] mVisible=true mFlags=[]}, InsetsSource: {3 mType=ime mFrame=[0,1491][1080,2400] mVisible=true mFlags=[]}, InsetsSource: {27 mType=displayCutout mFrame=[0,0][1080,80] mVisible=true mFlags=[]} }
10:57:24.200 ViewRootIm...oteEditor] com.ichi2.anki.debug                 I  MSG_WINDOW_FOCUS_CHANGED 0 0
10:57:24.202 ViewRootIm...oteEditor] com.ichi2.anki.debug                 I  MSG_WINDOW_FOCUS_CHANGED 0 0
10:57:24.212 Multimedia...ldActivity com.ichi2.anki.debug                 D  onCreateOptionsMenu() - mField.getType() = AUDIO_RECORDING
10:57:24.303 InputMethodManagerUtils com.ichi2.anki.debug                 D  startInputInner - Id : 0
10:57:24.303 InputMethodManager      com.ichi2.anki.debug                 I  startInputInner - IInputMethodManagerGlobalInvoker.startInputOrWindowGainedFocus
10:57:24.314 RemoteInpu...ectionImpl com.ichi2.anki.debug                 W  requestCursorAnchorInfo on inactive InputConnection
10:57:24.315 InsetsController        com.ichi2.anki.debug                 D  setRequestedVisibleTypes: visible=false, mask=ime, host=com.ichi2.anki.debug/com.ichi2.anki.NoteEditor, from=android.view.InsetsController.controlAnimationUnchecked:1424 android.view.InsetsController.applyAnimation:1979 android.view.InsetsController.applyAnimation:1942 android.view.InsetsController.hide:1382 android.view.ViewRootImpl$ViewRootHandler.handleMessageImpl:7116 android.view.ViewRootImpl$ViewRootHandler.handleMessage:7042 android.os.Handler.dispatchMessage:106 android.os.Looper.loopOnce:230 android.os.Looper.loop:319 android.app.ActivityThread.main:8893 
10:57:24.317 InsetsController        com.ichi2.anki.debug                 D  controlAnimationUncheckedInner: Added types=ime animType=1 host=com.ichi2.anki.debug/com.ichi2.anki.NoteEditor from=android.view.InsetsController.controlAnimationUnchecked:1428 android.view.InsetsController.applyAnimation:1979 android.view.InsetsController.applyAnimation:1942 
10:57:24.352 InsetsController        com.ichi2.anki.debug                 D  onStateChanged: host=com.ichi2.anki.debug/com.ichi2.anki.multimediacard.activity.MultimediaEditFieldActivity from=android.view.ViewRootImpl$ViewRootHandler.handleMessageImpl:7085 state=InsetsState: {mDisplayFrame=Rect(0, 0 - 1080, 2400), mDisplayCutout=DisplayCutout{insets=Rect(0, 80 - 0, 0) waterfall=Insets{left=0, top=0, right=0, bottom=0} boundingRect={Bounds=[Rect(0, 0 - 0, 0), Rect(510, 0 - 570, 80), Rect(0, 0 - 0, 0), Rect(0, 0 - 0, 0)]} cutoutPathParserInfo={CutoutPathParserInfo{displayWidth=1080 displayHeight=2400 physicalDisplayWidth=1080 physicalDisplayHeight=2400 density={3.0} cutoutSpec={M 0,0 H -10 V 26.66666666666667 H 10 V 0 H 0 Z @dp} rotation={0} scale={1.0} physicalPixelDisplaySizeRatio={1.0}}}}, mRoundedCorners=RoundedCorners{[RoundedCorner{position=TopLeft, radius=0, center=Point(0, 0)}, RoundedCorner{position=TopRight, radius=0, center=Point(0, 0)}, RoundedCorner{position=BottomRight, radius=0, center=Point(0, 0)}, RoundedCorner{position=BottomLeft, radius=0, center=Point(0, 0)}]}  mRoundedCornerFrame=Rect(0, 0 - 1080, 2400), mPrivacyIndicatorBounds=PrivacyIndicatorBounds {static bounds=Rect(948, 0 - 1080, 80) rotation=0}, mDisplayShape=DisplayShape{ spec=-1893857183 displayWidth=1080 displayHeight=2400 physicalPixelDisplaySizeRatio=1.0 rotation=0 offsetX=0 offsetY=0 scale=1.0}, mSources= { InsetsSource: {c06f0001 mType=navigationBars mFrame=[0,2355][1080,2400] mVisible=true mFlags=[SUPPRESS_SCRIM]}, InsetsSource: {c06f0004 mType=systemGestures mFrame=[0,0][90,2400] mVisible=true mFlags=[]}, InsetsSource: {c06f0005 mType=mandatorySystemGestures mFrame=[0,2304][1080,2400] mVisible=true mFlags=[]}, InsetsSource: {c06f0006 mType=tappableElement mFrame=[0,0][0,0] mVisible=true mFlags=[]}, InsetsSource: {c06f0024 mType=systemGestures mFrame=[990,0][1080,2400] mVisible=true mFlags=[]}, InsetsSource: {e9c60000 mType=statusBars mFrame=[0,0][1080,80] mVisible=true mFlags=[]}, InsetsSource: {e9c60005 mType=mandatorySystemGestures mFrame=[0,0][1080,116] mVisible=true mFlags=[]}, InsetsSource: {e9c60006 mType=tappableElement mFrame=[0,0][1080,80] mVisible=true mFlags=[]}, InsetsSource: {3 mType=ime mFrame=[0,1491][1080,2400] mVisible=false mFlags=[]}, InsetsSource: {27 mType=displayCutout mFrame=[0,0][1080,80] mVisible=true mFlags=[]} }
10:57:24.355 InsetsController        com.ichi2.anki.debug                 D  onStateChanged: host=com.ichi2.anki.debug/com.ichi2.anki.NoteEditor from=android.view.ViewRootImpl$ViewRootHandler.handleMessageImpl:7085 state=InsetsState: {mDisplayFrame=Rect(0, 0 - 1080, 2400), mDisplayCutout=DisplayCutout{insets=Rect(0, 80 - 0, 0) waterfall=Insets{left=0, top=0, right=0, bottom=0} boundingRect={Bounds=[Rect(0, 0 - 0, 0), Rect(510, 0 - 570, 80), Rect(0, 0 - 0, 0), Rect(0, 0 - 0, 0)]} cutoutPathParserInfo={CutoutPathParserInfo{displayWidth=1080 displayHeight=2400 physicalDisplayWidth=1080 physicalDisplayHeight=2400 density={3.0} cutoutSpec={M 0,0 H -10 V 26.66666666666667 H 10 V 0 H 0 Z @dp} rotation={0} scale={1.0} physicalPixelDisplaySizeRatio={1.0}}}}, mRoundedCorners=RoundedCorners{[RoundedCorner{position=TopLeft, radius=0, center=Point(0, 0)}, RoundedCorner{position=TopRight, radius=0, center=Point(0, 0)}, RoundedCorner{position=BottomRight, radius=0, center=Point(0, 0)}, RoundedCorner{position=BottomLeft, radius=0, center=Point(0, 0)}]}  mRoundedCornerFrame=Rect(0, 0 - 1080, 2400), mPrivacyIndicatorBounds=PrivacyIndicatorBounds {static bounds=Rect(948, 0 - 1080, 80) rotation=0}, mDisplayShape=DisplayShape{ spec=-1893857183 displayWidth=1080 displayHeight=2400 physicalPixelDisplaySizeRatio=1.0 rotation=0 offsetX=0 offsetY=0 scale=1.0}, mSources= { InsetsSource: {c06f0001 mType=navigationBars mFrame=[0,2355][1080,2400] mVisible=true mFlags=[SUPPRESS_SCRIM]}, InsetsSource: {c06f0004 mType=systemGestures mFrame=[0,0][90,2400] mVisible=true mFlags=[]}, InsetsSource: {c06f0005 mType=mandatorySystemGestures mFrame=[0,2304][1080,2400] mVisible=true mFlags=[]}, InsetsSource: {c06f0006 mType=tappableElement mFrame=[0,0][0,0] mVisible=true mFlags=[]}, InsetsSource: {c06f0024 mType=systemGestures mFrame=[990,0][1080,2400] mVisible=true mFlags=[]}, InsetsSource: {e9c60000 mType=statusBars mFrame=[0,0][1080,80] mVisible=true mFlags=[]}, InsetsSource: {e9c60005 mType=mandatorySystemGestures mFrame=[0,0][1080,116] mVisible=true mFlags=[]}, InsetsSource: {e9c60006 mType=tappableElement mFrame=[0,0][1080,80] mVisible=true mFlags=[]}, InsetsSource: {3 mType=ime mFrame=[0,1491][1080,2400] mVisible=true mFlags=[]}, InsetsSource: {27 mType=displayCutout mFrame=[0,0][1080,80] mVisible=true mFlags=[]} }
10:57:24.355 ImeTracker              com.ichi2.anki.debug                 I  com.ichi2.anki.debug:9d1e6f: onCancelled at PHASE_CLIENT_ANIMATION_CANCEL
10:57:24.355 InsetsController        com.ichi2.anki.debug                 D  cancelAnimation of types: 8, animType: 1, host: com.ichi2.anki.debug/com.ichi2.anki.NoteEditor, from:android.view.InsetsController.notifyControlRevoked:1765 android.view.InsetsSourceConsumer.setControl:154 android.view.ImeInsetsSourceConsumer.setControl:195 
10:57:24.356 ImeTracker              com.ichi2.anki.debug                 I  com.ichi2.anki.debug:b278c27e: onRequestHide at ORIGIN_CLIENT_HIDE_SOFT_INPUT reason HIDE_SOFT_INPUT_BY_INSETS_API
10:57:24.356 InputMethodManager_LC   com.ichi2.anki.debug                 I  notifyImeHidden: IInputMethodManagerGlobalInvoker.hideSoftInput
10:57:24.356 ImeTracker              com.ichi2.anki.debug                 I  com.ichi2.anki.debug:b278c27e: onFailed at PHASE_CLIENT_VIEW_SERVED
10:57:24.469 ViewRootIm...oteEditor] com.ichi2.anki.debug                 I  handleAppVisibility mAppVisible = true visible = false
10:57:24.469 ViewRootIm...w:8089e18] com.ichi2.anki.debug                 I  handleAppVisibility mAppVisible = true visible = false
10:57:24.490 InsetsController        com.ichi2.anki.debug                 D  onStateChanged: host=com.ichi2.anki.debug/com.ichi2.anki.NoteEditor from=android.view.ViewRootImpl.relayoutWindow:9940 state=InsetsState: {mDisplayFrame=Rect(0, 0 - 1080, 2400), mDisplayCutout=DisplayCutout{insets=Rect(0, 80 - 0, 0) waterfall=Insets{left=0, top=0, right=0, bottom=0} boundingRect={Bounds=[Rect(0, 0 - 0, 0), Rect(510, 0 - 570, 80), Rect(0, 0 - 0, 0), Rect(0, 0 - 0, 0)]} cutoutPathParserInfo={CutoutPathParserInfo{displayWidth=1080 displayHeight=2400 physicalDisplayWidth=1080 physicalDisplayHeight=2400 density={3.0} cutoutSpec={M 0,0 H -10 V 26.66666666666667 H 10 V 0 H 0 Z @dp} rotation={0} scale={1.0} physicalPixelDisplaySizeRatio={1.0}}}}, mRoundedCorners=RoundedCorners{[RoundedCorner{position=TopLeft, radius=0, center=Point(0, 0)}, RoundedCorner{position=TopRight, radius=0, center=Point(0, 0)}, RoundedCorner{position=BottomRight, radius=0, center=Point(0, 0)}, RoundedCorner{position=BottomLeft, radius=0, center=Point(0, 0)}]}  mRoundedCornerFrame=Rect(0, 0 - 1080, 2400), mPrivacyIndicatorBounds=PrivacyIndicatorBounds {static bounds=Rect(948, 0 - 1080, 80) rotation=0}, mDisplayShape=DisplayShape{ spec=-1893857183 displayWidth=1080 displayHeight=2400 physicalPixelDisplaySizeRatio=1.0 rotation=0 offsetX=0 offsetY=0 scale=1.0}, mSources= { InsetsSource: {c06f0001 mType=navigationBars mFrame=[0,2355][1080,2400] mVisible=true mFlags=[SUPPRESS_SCRIM]}, InsetsSource: {c06f0004 mType=systemGestures mFrame=[0,0][90,2400] mVisible=true mFlags=[]}, InsetsSource: {c06f0005 mType=mandatorySystemGestures mFrame=[0,2304][1080,2400] mVisible=true mFlags=[]}, InsetsSource: {c06f0006 mType=tappableElement mFrame=[0,0][0,0] mVisible=true mFlags=[]}, InsetsSource: {c06f0024 mType=systemGestures mFrame=[990,0][1080,2400] mVisible=true mFlags=[]}, InsetsSource: {e9c60000 mType=statusBars mFrame=[0,0][1080,80] mVisible=true mFlags=[]}, InsetsSource: {e9c60005 mType=mandatorySystemGestures mFrame=[0,0][1080,116] mVisible=true mFlags=[]}, InsetsSource: {e9c60006 mType=tappableElement mFrame=[0,0][1080,80] mVisible=true mFlags=[]}, InsetsSource: {3 mType=ime mFrame=[0,1491][1080,2400] mVisible=false mFlags=[]}, InsetsSource: {27 mType=displayCutout mFrame=[0,0][1080,80] mVisible=true mFlags=[]} }
10:57:24.492 ViewRootIm...oteEditor] com.ichi2.anki.debug                 I  Relayout returned: old=(0,0,1080,2400) new=(0,0,1080,2400) req=(1080,2400)8 dur=6 res=0x402 s={false 0x0} ch=true seqId=0
10:57:24.516 ViewRootIm...oteEditor] com.ichi2.anki.debug                 I  stopped(true) old = false
10:57:24.516 ViewRootIm...oteEditor] com.ichi2.anki.debug                 D  WindowStopped on com.ichi2.anki.debug/com.ichi2.anki.NoteEditor set to true
10:57:24.517 AnkiDroidApp$onCreate   com.ichi2.anki.debug                 I  NoteEditor::onStop
10:57:24.518 WidgetStatus            com.ichi2.anki.debug                 D  WidgetStatus.update(): already running or not enabled
10:57:24.518 NoteEditor              com.ichi2.anki.debug                 I  Saving instance
10:57:24.520 AnkiDroidApp$onCreate   com.ichi2.anki.debug                 I  NoteEditor::onSaveInstanceState
10:57:24.524 ViewRootIm...oteEditor] com.ichi2.anki.debug                 I  Relayout returned: old=(0,0,1080,2400) new=(0,0,1080,2400) req=(1080,2400)8 dur=2 res=0x402 s={false 0x0} ch=false seqId=0
10:57:24.537 InsetsController        com.ichi2.anki.debug                 D  onStateChanged: host=com.ichi2.anki.debug/com.ichi2.anki.multimediacard.activity.MultimediaEditFieldActivity from=android.view.ViewRootImpl$ViewRootHandler.handleMessageImpl:7072 state=InsetsState: {mDisplayFrame=Rect(0, 0 - 1080, 2400), mDisplayCutout=DisplayCutout{insets=Rect(0, 80 - 0, 0) waterfall=Insets{left=0, top=0, right=0, bottom=0} boundingRect={Bounds=[Rect(0, 0 - 0, 0), Rect(510, 0 - 570, 80), Rect(0, 0 - 0, 0), Rect(0, 0 - 0, 0)]} cutoutPathParserInfo={CutoutPathParserInfo{displayWidth=1080 displayHeight=2400 physicalDisplayWidth=1080 physicalDisplayHeight=2400 density={3.0} cutoutSpec={M 0,0 H -10 V 26.66666666666667 H 10 V 0 H 0 Z @dp} rotation={0} scale={1.0} physicalPixelDisplaySizeRatio={1.0}}}}, mRoundedCorners=RoundedCorners{[RoundedCorner{position=TopLeft, radius=0, center=Point(0, 0)}, RoundedCorner{position=TopRight, radius=0, center=Point(0, 0)}, RoundedCorner{position=BottomRight, radius=0, center=Point(0, 0)}, RoundedCorner{position=BottomLeft, radius=0, center=Point(0, 0)}]}  mRoundedCornerFrame=Rect(0, 0 - 1080, 2400), mPrivacyIndicatorBounds=PrivacyIndicatorBounds {static bounds=Rect(948, 0 - 1080, 80) rotation=0}, mDisplayShape=DisplayShape{ spec=-1893857183 displayWidth=1080 displayHeight=2400 physicalPixelDisplaySizeRatio=1.0 rotation=0 offsetX=0 offsetY=0 scale=1.0}, mSources= { InsetsSource: {c06f0001 mType=navigationBars mFrame=[0,2355][1080,2400] mVisible=true mFlags=[SUPPRESS_SCRIM]}, InsetsSource: {c06f0004 mType=systemGestures mFrame=[0,0][90,2400] mVisible=true mFlags=[]}, InsetsSource: {c06f0005 mType=mandatorySystemGestures mFrame=[0,2304][1080,2400] mVisible=true mFlags=[]}, InsetsSource: {c06f0006 mType=tappableElement mFrame=[0,0][0,0] mVisible=true mFlags=[]}, InsetsSource: {c06f0024 mType=systemGestures mFrame=[990,0][1080,2400] mVisible=true mFlags=[]}, InsetsSource: {e9c60000 mType=statusBars mFrame=[0,0][1080,80] mVisible=true mFlags=[]}, InsetsSource: {e9c60005 mType=mandatorySystemGestures mFrame=[0,0][1080,116] mVisible=true mFlags=[]}, InsetsSource: {e9c60006 mType=tappableElement mFrame=[0,0][1080,80] mVisible=true mFlags=[]}, InsetsSource: {3 mType=ime mFrame=[0,0][0,0] mVisible=false mFlags=[]}, InsetsSource: {27 mType=displayCutout mFrame=[0,0][1080,80] mVisible=true mFlags=[]} }
10:57:24.542 InsetsController        com.ichi2.anki.debug                 D  onStateChanged: host=com.ichi2.anki.debug/com.ichi2.anki.multimediacard.activity.MultimediaEditFieldActivity from=android.view.ViewRootImpl$ViewRootHandler.handleMessageImpl:7072 state=InsetsState: {mDisplayFrame=Rect(0, 0 - 1080, 2400), mDisplayCutout=DisplayCutout{insets=Rect(0, 80 - 0, 0) waterfall=Insets{left=0, top=0, right=0, bottom=0} boundingRect={Bounds=[Rect(0, 0 - 0, 0), Rect(510, 0 - 570, 80), Rect(0, 0 - 0, 0), Rect(0, 0 - 0, 0)]} cutoutPathParserInfo={CutoutPathParserInfo{displayWidth=1080 displayHeight=2400 physicalDisplayWidth=1080 physicalDisplayHeight=2400 density={3.0} cutoutSpec={M 0,0 H -10 V 26.66666666666667 H 10 V 0 H 0 Z @dp} rotation={0} scale={1.0} physicalPixelDisplaySizeRatio={1.0}}}}, mRoundedCorners=RoundedCorners{[RoundedCorner{position=TopLeft, radius=0, center=Point(0, 0)}, RoundedCorner{position=TopRight, radius=0, center=Point(0, 0)}, RoundedCorner{position=BottomRight, radius=0, center=Point(0, 0)}, RoundedCorner{position=BottomLeft, radius=0, center=Point(0, 0)}]}  mRoundedCornerFrame=Rect(0, 0 - 1080, 2400), mPrivacyIndicatorBounds=PrivacyIndicatorBounds {static bounds=Rect(948, 0 - 1080, 80) rotation=0}, mDisplayShape=DisplayShape{ spec=-1893857183 displayWidth=1080 displayHeight=2400 physicalPixelDisplaySizeRatio=1.0 rotation=0 offsetX=0 offsetY=0 scale=1.0}, mSources= { InsetsSource: {c06f0001 mType=navigationBars mFrame=[0,2355][1080,2400] mVisible=true mFlags=[SUPPRESS_SCRIM]}, InsetsSource: {c06f0004 mType=systemGestures mFrame=[0,0][90,2400] mVisible=true mFlags=[]}, InsetsSource: {c06f0005 mType=mandatorySystemGestures mFrame=[0,2304][1080,2400] mVisible=true mFlags=[]}, InsetsSource: {c06f0006 mType=tappableElement mFrame=[0,0][0,0] mVisible=true mFlags=[]}, InsetsSource: {c06f0024 mType=systemGestures mFrame=[990,0][1080,2400] mVisible=true mFlags=[]}, InsetsSource: {e9c60000 mType=statusBars mFrame=[0,0][1080,80] mVisible=true mFlags=[]}, InsetsSource: {e9c60005 mType=mandatorySystemGestures mFrame=[0,0][1080,116] mVisible=true mFlags=[]}, InsetsSource: {e9c60006 mType=tappableElement mFrame=[0,0][1080,80] mVisible=true mFlags=[]}, InsetsSource: {3 mType=ime mFrame=[0,0][0,0] mVisible=false mFlags=[]}, InsetsSource: {27 mType=displayCutout mFrame=[0,0][1080,80] mVisible=true mFlags=[]} }
10:57:25.556 AudioRecor...Controller com.ichi2.anki.debug                 I  primary 'record' button clicked
10:57:25.556 AudioRecor...Controller com.ichi2.anki.debug                 I  starting recording
10:57:25.556 MediaRecorder           com.ichi2.anki.debug                 D  Constructor MediaRecorder
10:57:25.557 MediaRecorderJNI        com.ichi2.anki.debug                 I  setup
10:57:25.561 MediaRecorderJNI        com.ichi2.anki.debug                 I  setAudioSource(1)
10:57:25.574 MediaRecorderJNI        com.ichi2.anki.debug                 I  setAudioEncoder(3)
10:57:25.574 MediaRecorderJNI        com.ichi2.anki.debug                 I  setParameter(), params: audio-param-number-of-channels=2
10:57:25.574 MediaRecorderJNI        com.ichi2.anki.debug                 I  setParameter(), params: audio-param-sampling-rate=44100
10:57:25.574 MediaRecorderJNI        com.ichi2.anki.debug                 I  setParameter(), params: audio-param-encoding-bitrate=192000
10:57:25.575 MediaRecorderJNI        com.ichi2.anki.debug                 I  setOutputFile
10:57:25.575 MediaRecorderJNI        com.ichi2.anki.debug                 I  prepare
10:57:25.578 MediaRecorder           com.ichi2.anki.debug                 I  prepare elapsed time : 3 ms
10:57:25.580 MediaRecorderJNI        com.ichi2.anki.debug                 I  start
10:57:25.660 AudioRecor...Controller com.ichi2.anki.debug                 I  ui: AppendToRecording::CLEARED -> AppendToRecording::RECORDING_IN_PROGRESS
10:57:29.349 AudioRecor...Controller com.ichi2.anki.debug                 I  primary 'record' button clicked
10:57:29.349 AudioRecor...Controller com.ichi2.anki.debug                 I  pausing recording
10:57:29.349 MediaRecorderJNI        com.ichi2.anki.debug                 I  pause
10:57:29.350 AudioRecor...Controller com.ichi2.anki.debug                 I  ui: AppendToRecording::RECORDING_IN_PROGRESS -> AppendToRecording::RECORDING_PAUSED
10:57:29.982 Multimedia...ldActivity com.ichi2.anki.debug                 I  Save button pressed
10:57:29.983 MediaRecorderJNI        com.ichi2.anki.debug                 I  stop
10:57:30.042 AnkiActivity            com.ichi2.anki.debug                 I  finishWithAnimation DEFAULT
10:57:30.049 AnkiDroidApp$onCreate   com.ichi2.anki.debug                 I  MultimediaEditFieldActivity::onPause
10:57:30.049 AudioRecor...Controller com.ichi2.anki.debug                 I  activity paused: stopping recording/resetting player
10:57:30.051 AudioRecor...Controller com.ichi2.anki.debug                 I  ui: AppendToRecording::RECORDING_PAUSED -> AppendToRecording::CLEARED
10:57:30.055 MediaRecorderJNI        com.ichi2.anki.debug                 I  stop
10:57:30.055 MediaRecorder           com.ichi2.anki.debug                 E  stop called in an invalid state: 1
10:57:30.055 AndroidRuntime          com.ichi2.anki.debug                 D  Shutting down VM
10:57:30.058 AndroidRuntime          com.ichi2.anki.debug                 E  FATAL EXCEPTION: main
                                                                             Process: com.ichi2.anki.debug, PID: 24494
                                                                             java.lang.RuntimeException: Unable to pause activity {com.ichi2.anki.debug/com.ichi2.anki.multimediacard.activity.MultimediaEditFieldActivity}: java.lang.IllegalStateException
                                                                             	at android.app.ActivityThread.performPauseActivityIfNeeded(ActivityThread.java:5779)
                                                                             	at android.app.ActivityThread.performPauseActivity(ActivityThread.java:5730)
                                                                             	at android.app.ActivityThread.handlePauseActivity(ActivityThread.java:5682)
                                                                             	at android.app.servertransaction.PauseActivityItem.execute(PauseActivityItem.java:48)
                                                                             	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
                                                                             	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:180)
                                                                             	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:98)
                                                                             	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2685)
                                                                             	at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                             	at android.os.Looper.loopOnce(Looper.java:230)
                                                                             	at android.os.Looper.loop(Looper.java:319)
                                                                             	at android.app.ActivityThread.main(ActivityThread.java:8893)
                                                                             	at java.lang.reflect.Method.invoke(Native Method)
                                                                             	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:608)
                                                                             	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
                                                                             Caused by: java.lang.IllegalStateException
                                                                             	at android.media.MediaRecorder._stop(Native Method)
                                                                             	at android.media.MediaRecorder.stop(MediaRecorder.java:1628)
                                                                             	at com.ichi2.anki.multimediacard.AudioRecorder.stopRecording(AudioRecorder.kt:78)
                                                                             	at com.ichi2.audio.AudioRecordingController.clearRecording(AudioRecordingController.kt:598)
                                                                             	at com.ichi2.audio.AudioRecordingController.onViewFocusChanged(AudioRecordingController.kt:282)
                                                                             	at com.ichi2.audio.AudioRecordingController$createUI$8$1.onActivityPaused(AudioRecordingController.kt:246)
                                                                             	at android.app.Application.dispatchActivityPaused(Application.java:480)
                                                                             	at android.app.Activity.dispatchActivityPaused(Activity.java:1552)
                                                                             	at android.app.Activity.onPause(Activity.java:2529)
                                                                             	at androidx.fragment.app.FragmentActivity.onPause(FragmentActivity.java:282)
                                                                             	at android.app.Activity.performPause(Activity.java:9150)
                                                                             	at android.app.Instrumentation.callActivityOnPause(Instrumentation.java:1657)
                                                                             	at android.app.ActivityThread.performPauseActivityIfNeeded(ActivityThread.java:5769)
                                                                             	at android.app.ActivityThread.performPauseActivity(ActivityThread.java:5730) 
                                                                             	at android.app.ActivityThread.handlePauseActivity(ActivityThread.java:5682) 
                                                                             	at android.app.servertransaction.PauseActivityItem.execute(PauseActivityItem.java:48) 
                                                                             	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45) 
                                                                             	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:180) 
                                                                             	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:98) 
                                                                             	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2685) 
                                                                             	at android.os.Handler.dispatchMessage(Handler.java:106) 
                                                                             	at android.os.Looper.loopOnce(Looper.java:230) 
                                                                             	at android.os.Looper.loop(Looper.java:319) 
                                                                             	at android.app.ActivityThread.main(ActivityThread.java:8893) 
                                                                             	at java.lang.reflect.Method.invoke(Native Method) 
                                                                             	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:608) 
                                                                             	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103) 

```

</details>

--- a/AnkiDroid/src/main/java/com/ichi2/audio/AudioRecordingController.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/audio/AudioRecordingController.kt
@@ -43,6 +43,7 @@ import com.ichi2.anki.showThemedToast
 import com.ichi2.anki.snackbar.showSnackbar
 import com.ichi2.anki.utils.elapsed
 import com.ichi2.anki.utils.formatAsString
+import com.ichi2.annotations.NeedsTest
 import com.ichi2.audio.AudioRecordingController.RecordingState.AppendToRecording
 import com.ichi2.audio.AudioRecordingController.RecordingState.ImmediatePlayback
 import com.ichi2.compat.CompatHelper
@@ -276,6 +277,8 @@ class AudioRecordingController :
         }
     }
 
+    /** Called on pause, and when 'done' is pressed */
+    @NeedsTest("16321: record -> 'done' without pressing save")
     fun onViewFocusChanged() {
         Timber.i("activity paused: stopping recording/resetting player")
         if (isRecording || isRecordingPaused) {
@@ -434,7 +437,6 @@ class AudioRecordingController :
     fun toggleSave(vibrate: Boolean = true) {
         Timber.i("recording completed")
         if (vibrate) CompatHelper.compat.vibrate(context, 20)
-        setUiState(state.afterSave())
         stopAndSaveRecording()
         // show this snackbar only in the edit field/multimedia activity
         if (inEditField) (context as Activity).showSnackbar(context.resources.getString(R.string.audio_saved))
@@ -571,6 +573,7 @@ class AudioRecordingController :
         }
         isRecording = false
         isAudioRecordingSaved = true
+        setUiState(state.afterSave())
         // save recording only in the edit field not in the reviewer but save it temporarily
         if (inEditField) saveRecording()
     }

